### PR TITLE
Link - new window fixes

### PR DESCRIPTION
**Pull request by bearfriend**
- Match the host `white-space` style
- Don't underline the space between the link text and icon

The only downside is that it doesn't update with changes to the host `white-space`, but that should basically not be an issue in practice.

Alternative to the alternatives #4050 and #4045

### CODE REVIEW

File: components/link/link.js
Comment (dlockhart) [inline components/link/link.js:184]: I think we can make a slight optimization and apply `white-space: nowrap;` to the `<a>`, removing one `<span>`?

File: components/link/link.js
Comment (dlockhart) [inline components/link/link.js:128]: When I asked Jeff how much space he'd like between the link text and the icon, he said:

```suggestion
					margin-inline-start: 6px;
```

**Review by dlockhart — APPROVED** [review]
Cool, that's a really creative solution that I definitely wouldn't have thought of! Seems like the best of all worlds?

File: components/link/link.js
Comment (bearfriend) [reply components/link/link.js:128]: Doesn't it seem weird for it to be fixed though?

Comment (dlockhart) [reply components/link/link.js:128]: Maybe. This ends up being `5.51px` @ `20px` default font size, so I'm good with it either way.

Comment (bearfriend) [reply components/link/link.js:128]: If we make the icon:
```
width: 0.78em;
height: 0.78em;
margin-inline-start: 0.33em; /* 6px default */
```

Then the icon scales with the link no what people do to it.

Comment (dlockhart) [reply components/link/link.js]: Chatted with Jeff -- he definitely likes the idea of scaling the icon with the text like we do with subtle buttons, so good call.

He'd like for it to be `18px` for "normal" links at default font size, so `0.9em` should achieve that.

For `small` links, they're a bit weird in that the font-size is `rem`-based and not `em`-based, so the font size never actually goes up/down with the cascaded font-size. So for them, I think we want `0.7rem` which would result in `14px`.

Does that make sense?

**Review by NicholasHallman — APPROVED** [review]
Nice! LGTM

File: components/link/link.js
Comment (bearfriend) [reply components/link/link.js]: It does except I still think `1em` for the small links makes more sense because it shouldn't really care about anything else at that point except the font-size of the thing it lives inside. So, whether the small link were to change its font-size to `0.8rem`, '16px', or `0.842105263156em`, the icon would still be fine.

Comment (dlockhart) [reply components/link/link.js]: Ah yep cool cool, carry on 😎

Comment (dlockhart) [reply components/link/link.js]: Could we just roll with `0.95em` for both then for ultimate consistency and less CSS?

File: components/link/link.js
Comment (dlockhart) [inline components/link/link.js:167]: Since these get put inline, I think typically we'd only put dynamic stuff in these style maps? vs. just putting it up in the CSS?

Comment (bearfriend) [reply components/link/link.js:167]: lol, yes, of course.

**Review by dlockhart — APPROVED** [review]

**Comment by bearfriend** [comment]
Finally figured out how to avoid the alternating/inherited `white-space` styles. No more `getComputedStyle`, which wasnt guaranteed to be correct on first render.

**Comment by bearfriend** [comment]
This all looks god to me now. `vdiff` changes look good, but I'll let those run and merge on Nick's PR. The one `visual-diff` change is a flake, I'm pretty sure. I'll merge in the morning unless I see objections.

**Comment by dlockhart** [comment]
Is it weird that the two "new window" print vdiffs grew in width by 8-10 pixels? Maybe not enough stuff is getting hidden in print mode?

**Comment by bearfriend** [comment]
> Is it weird that the two "new window" print vdiffs grew in width by 8-10 pixels? Maybe not enough stuff is getting hidden in print mode?

They actually shrunk by 8-10px because the space/margin is being properly hidden in print mode now.

**Review by dlockhart — APPROVED** [review]

**Pull request merged**
